src/learning/gmm.rs

```diff
@@ -302,7 +303,7 @@ impl GaussianMixtureModel {
         self.model_covars = Some(new_covs);
     }
 
-    fn compute_cov(&self, diff: Matrix<f64>, weight: f64) -> Matrix<f64> {
+    fn compute_cov(&self, diff: &Matrix<f64>, weight: f64) -> Matrix<f64> {
```

Comment:
It doesn't make a difference in this case but it may in the future. For example we _could_ do in-place operations for transpose or `elemul`. It's not happening right now though so no reason to change this.